### ISSUE TITLE
503 Service Unavailable

### ISSUE THREAD

**Issue by TheNugg** [geerlingguy/drupal-vm#518]
No matter what I try, I keep getting a 503 error when trying to navigate to my Drupal site.

I can get to the dashboard no problem and I can get to MailHog on port 8052, but I cannot get to anything that seems to be running on port 80.

I have tried provisioning with Varnish and without Varnish; both with the same result.

Any thoughts?

**Comment by TheNugg** [comment]
I should mentioned that the above is when using Apache.

I have since switched to nginx and I am now getting: 502 Bad Gateway.

I feel silly with troubleshooting these issues, but I cannot seem to find the culprit.

**Comment by codethirteen** [comment]
It sounds like PHP is not running, but Apache and Nginx are running. I would SSH into the box and check which services are running first:   

    vagrant ssh
    service --status-all

**Comment by geerlingguy** [comment]
Yeah, either that or one of the other things that needs to be running is in a bad state. Try logging in with `vagrant ssh` then running:

```
sudo service apache2 restart
sudo service php5-fpm restart
sudo service mysql restart
sudo service varnish restart
```

**Comment by geerlingguy** [comment]
Going to close this out for now; let me know if it's still not working!

**Comment by 4hmedSamir** [comment]
I'm getting the same error, list of services:
[ + ]  apache-htcacheclean
 [ + ]  apache2
 [ + ]  apparmor
 [ - ]  bootmisc.sh
 [ - ]  checkfs.sh
 [ - ]  checkroot-bootclean.sh
 [ - ]  checkroot.sh
 [ + ]  console-setup
 [ + ]  cron
 [ - ]  cryptdisks
 [ - ]  cryptdisks-early
 [ + ]  dbus
 [ - ]  firewall
 [ + ]  grub-common
 [ - ]  hostname.sh
 [ - ]  hwclock.sh
 [ + ]  irqbalance
 [ + ]  keyboard-setup
 [ - ]  killprocs
 [ + ]  kmod
 [ - ]  lvm2
 [ + ]  lvm2-lvmetad
 [ + ]  lvm2-lvmpolld
 [ - ]  mailhog
 [ + ]  memcached
 [ - ]  mountall-bootclean.sh
 [ - ]  mountall.sh
 [ - ]  mountdevsubfs.sh
 [ - ]  mountkernfs.sh
 [ - ]  mountnfs-bootclean.sh
 [ - ]  mountnfs.sh
 [ + ]  mysql
 [ + ]  networking
 [ + ]  nfs-kernel-server
 [ + ]  ondemand
 [ + ]  php7.1-fpm
 [ - ]  plymouth
 [ - ]  plymouth-log
 [ + ]  postfix
 [ + ]  procps
 [ + ]  rc.local
 [ + ]  resolvconf
 [ + ]  rpcbind
 [ - ]  rsync
 [ + ]  rsyslog
 [ - ]  sendsigs
 [ + ]  ssh
 [ + ]  udev
 [ - ]  ufw
 [ - ]  umountfs
 [ - ]  umountnfs.sh
 [ - ]  umountroot
 [ - ]  unattended-upgrades
 [ + ]  urandom
 [ - ]  uuidd
 [ + ]  varnish
 [ + ]  varnishlog
 [ + ]  varnishncsa
 [ - ]  x11-common

**Comment by geerlingguy** [comment]
@4hmedSamir - Are you using Drupal VM? There's a lot of services in there that look like they aren't part of Drupal VM's default set of tools.

**Comment by 4hmedSamir** [comment]
Yes, I'm using drupal-vm-4.4.1-0. I think PHP-FPM  is not executing.

**Comment by oxyc** [comment]
@4hmedSamir would you mind opening up a new issue and post your config.yml

**Comment by 4hmedSamir** [comment]
@oxyc  Here is the new issue [1285](https://github.com/geerlingguy/drupal-vm/issues/1285)

**Comment by narendradesai** [comment]
@TheNugg Even I was facing the same issue. I was not able access Drupal website, it was throwing `503 error`. Here is the error log:
`tail -f /var/log/apache2/error.log`
**Error:**
`[Wed Nov 13 14:50:56.059346 2019] [proxy:error] [pid 13918:tid 140227166623488] (111)Connection refused: AH00957: FCGI: attempt to connect to 127.0.0.1:9000 (*) failed
[Wed Nov 13 14:50:56.060439 2019] [proxy_fcgi:error] [pid 13918:tid 140227166623488] [client 192.168.173.1:57787] AH01079: failed to make connection to backend: 127.0.0.1
[Wed Nov 13 14:56:48.732865 2019] [mpm_event:notice] [pid 13915:tid 140227268401024] AH00491: caught SIGTERM, shutting down`

**Reason:** `apache` was not able to make connection with `127.0.0.1:9000.`

**Solution:**
**_Step 1:_** Check if `PHP-FPM` is running.
I checked the php vi /etc/php/7.3/fpm/pool.d/www.conf file to see what socket listening to in the listen line: `listen = /run/php/php7.3-fpm.sock`

**_Step2:_** Then add that in `vi /etc/apache2/sites-available/vhost.conf`
`<FilesMatch \.php$>
  SetHandler "proxy:unix:/run/php/php7.3-fpm.sock|fcgi://127.0.0.1:9000"
</FilesMatch>`
   **_Example.:_**
   <VirtualHost *:80>
  ServerName dashboard.local.example.com
  ServerAlias www.dashboard.local.example.com
  DocumentRoot "/var/www/dashboard"
  <Directory "/var/www/dashboard">
    AllowOverride All
    Options -Indexes +FollowSymLinks
    Require all granted
  </Directory>
  <FilesMatch \.php$>
  SetHandler "proxy:unix:/run/php/php7.3-fpm.sock|fcgi://127.0.0.1:9000"
<\/FilesMatch>
<\/VirtualHost>`

**_Step3:_** `sudo service apache2 restart`

**Comment by wilfi** [comment]
Thanks @narendradesai . That worked for me.
Initially on `vagrant reload --provision` I got the **chgrp failed** error on 
`{{ php_fpm_conf_path }}/pool.d` . Something similar to [this issue](https://github.com/geerlingguy/ansible-role-nodejs/issues/91). 
Not sure if both are related.

**Comment by jeremdow** [comment]
> <FilesMatch \.php$> SetHandler "proxy:unix:/run/php/php7.3-fpm.sock|fcgi://127.0.0.1:9000" </FilesMatch>

Yes, or just set PHP-FPM back to listening on the port.

/etc/php/7.3/fpm/pool.d/www.conf
`listen = 9000`

@geerlingguy it the socket the default, and per @wilfi above step is failing to provision the port?
I just noticed this after rebuilding my vm today, with no change to the config.

**Comment by scg-emendoza** [comment]
I received a similar 503 Service Unavailable message after a complete fresh install on MacOS Catalina. 
I got my local Drupal site working by updating the php www.conf file (inside vagrant :: /etc/php/7.3/fpm/pool.d/www.conf),

adding:

**listen = 127.0.0.1:9000**

Followed the instructions from this post:
 https://github.com/fideloper/Vaprobash/issues/437

I hope this helps.

**Comment by kovacskri** [comment]
I had the same issue on Arch Linux host OS.

The solution was to run `sudo a2enmod proxy_fcgi` add `listen = 127.0.0.1:9000` to `/etc/php/7.2/fpm/pool.d/www.conf`.

**Comment by seanr** [comment]
It seems like setting `php_fpm_listen: "127.0.0.1:9000"` in box/config.yml updates the vhosts but not /etc/php/7.2/fpm/pool.d/www.conf - is there somewhere else that needs to be set (or some other config option)?

**Comment by cosmicdreams** [comment]
I've tried this solution and it worked for me.  I previously thought it was the upgrade to Catalina 10.15.2 that broken my setup.  But now that this solution works, I'm suspicious of the upgrade to 1.3.1 of this vagrant box. 

Can you help me track down the proper place to report this bug so a fix can be spread out properly?

**Comment by alex-moreno** [comment]
@narendradesai comment worked for me

**Comment by apparatchiki** [comment]
Catalina 10.15.7, though I dont think the host is specially significant, a long running blt vm -drupal vm based- was failing vagrant up install first and resulting on a http 503 later. Install problems included having drupal console on the extras section, the drupal console installer was down with a http 500. Long story short: these were the succession of fixes implemented, in case this helps anyone:

1. On the box/config.yml, while updating from php 7.2 to 7.3, be sure to upgrade php-xdebug with it to the last version, we went from 2.6.0 to 2.9.8, this was another problem provisioning the vm.

2. Drupal Console seems to be getting less and less attention these days, the installer was offline and blocking the vm provision, we removed it from the extras section.

3. PHP-FPM was failing to connect as described by @seanr, be sure to have the php_fpm_listen: "127.0.0.1:9000" setting in the config.yml, for us this was enough to solve the 503 status.
